sentinel/src/lib/feed.js

```diff
@@ -38,6 +63,94 @@ const registerFeed = seq => {
     });
 };
 
+const deleteReadDocs = change => {
+  // we don't know if the deleted doc was a report or a message so
+  // attempt to delete both
+  const possibleReadDocIds = ['report', 'message'].map(
+    type => `read:${type}:${change.id}`
+  );
+
+  return db.allDbs().then(dbs => {
+    const userDbs = dbs.filter(dbName => dbName.indexOf(`${db.medicDbName}-user-`) === 0);
+    return userDbs.reduce((p, userDb) => {
+      return p.then(() => {
+        const metaDb = db.get(userDb);
+        return metaDb
+          .allDocs({ keys: possibleReadDocIds })
+          .then(results => {
+            const row = results.rows.find(row => !row.error);
+            if (!row) {
+              return;
+            }
+
+            return metaDb.remove(row.id, row.value.rev).catch(err => {
+              // ignore 404s or 409s - the doc was probably deleted client side already
+              if (err && err.status !== 404 && err.status !== 409) {
+                throw err;
+              }
+            });
+          })
+          .then(() => {
+            db.close(metaDb);
+          })
+          .catch(err => {
+            db.close(metaDb);
+            throw err;
+          });
+      });
+    }, Promise.resolve());
+
+  });
+};
+
+
+const changeQueue = async.queue((change, callback) => {
+  if (lastSeq && change.seq === lastSeq) {
+    lastSeq = undefined;
```

Comment:
I'm not seeing the logic here. Why would you mark the lastSeq as undefined? Is it no longer the last seq?